src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -56,45 +55,50 @@ public class PackageSourceProvider : IPackageSourceProvider
             _configurationDefaultSources = LoadConfigurationDefaultSources(configurationDefaultSources);
         }
 
-        private static IEnumerable<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
+        private static IReadOnlyList<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
         {
 #if !IS_CORECLR
             // Global default NuGet source doesn't make sense on Mono
-            if (RuntimeEnvironmentHelper.IsMono)
+            if (Common.RuntimeEnvironmentHelper.IsMono)
             {
-                return Enumerable.Empty<PackageSource>();
+                return Array.Empty<PackageSource>();
             }
 #endif
             var packageSourceLookup = new Dictionary<string, IndexedPackageSource>(StringComparer.OrdinalIgnoreCase);
             var packageIndex = 0;
 
             foreach (var packageSource in configurationDefaultSources)
             {
-                packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource, packageSource.Name);
+                packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource);
```

Comment:
The `AddOrUpdateIndexedSource`` method updates` the `packageIndex` and returns the new value. Could we consider changing the parameter to `ref`? This suggestion might be beyond the scope of this PR, but the current usage of `packageIndex` is a bit confusing without navigating into the `AddOrUpdateIndexedSource` method.

Reply:
Thank you for addressing this feedback in https://github.com/NuGet/NuGet.Client/pull/5516/commits/0719477b182093e376462bb2d8983c336fda6a39 commit.